viz_scripts/survey_responses.ipynb

```diff
@@ -0,0 +1,343 @@
+    "print(\"Options dictionary:\\n\", opt_dict)\n",
+    "\n",
+    "#color dictionary\n",
+    "color_map = scaffolding.mapping_color_surveys(opt_dict)\n",
+    "\n",
+    "#format survey trips into responses dataframe\n",
+    "df_responses = create_dataframe(survey_trips)\n",
+    "file_suffix = scaffolding.get_file_suffix(year, month, program)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93dbaa5e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#create the total dfs\n",
+    "total_dfs = {}\n",
+    "#for conditional surveys!\n",
+    "wrapped_config = {'survey_info': survey_info}\n",
+    "if 'buttons' in survey_info.keys():\n",
+    "    all_composite_trips['survey_name_prompted'] = all_composite_trips.apply(lambda row: conditional_surveys.survey_prompted_for_trip(row.to_dict(), wrapped_config), axis=1)\n",
+    "    \n",
+    "    for survey_name in list(sheet_list.keys()):\n",
+    "        if survey_name in all_composite_trips['survey_name_prompted'].unique():\n",
+    "            total_dfs[survey_name] = all_composite_trips[all_composite_trips['survey_name_prompted'] == survey_name]\n",
+    "        else:\n",
+    "            #never prompted\n",
+    "            total_dfs[survey_name] = pd.DataFrame()\n",
+    "else:\n",
+    "    survey_name = list(sheet_list.keys())[0] #there is only one if non-conditional\n",
+    "    total_dfs[survey_name] = all_composite_trips\n",
+    "    \n",
+    "for key in total_dfs.keys():\n",
+    "    print(key, \":\", len(total_dfs[key]), \"trips\", total_dfs[key].user_id.nunique(), \"users\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7bcebeef",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "#create one plot per question in the survey\n",
+    "for survey_name in quest_dict.keys():\n",
+    "    print(\"Charts for:\", survey_name)\n",
+    "    debug_df = generate_debug_df(program, include_test_users, total_dfs[survey_name], total_dfs[survey_name][total_dfs[survey_name]['user_input'] != {} if len(total_dfs[survey_name]) > 0 else pd.DataFrame()])\n",
+    "    \n",
+    "\n",
+    "    for col in quest_dict[survey_name].keys():\n",
+    "        \n",
+    "        filename = col + file_suffix\n",
+    "        plot_title_no_quality = quest_dict[survey_name][col]\n",
+    "\n",
+    "        try:\n",
+    "            quest_frame = df_responses.copy()\n",
+    "            quest_frame = quest_frame[quest_frame[col] != \"\"]\n",
+    "            qual_text = scaffolding.get_quality_text(total_dfs[survey_name], quest_frame, mode_of_interest, include_test_users)\n",
+    "            plot_title = plot_title_no_quality+'\\n'+qual_text\n",
+    "            \n",
+    "            labels = quest_frame[col].value_counts(dropna=True).keys().tolist()\n",
+    "            labels = traslate_options(quest_frame[col].value_counts(dropna=True).keys().tolist(), opt_dict)\n",
```

Comment:
isn't this just

```
labels = translate_options(labels, opt_dict)
```

Reply:
Also, as expected, this is more complicated than the pre-processing that I have rolled in to `plot_and_text_stacked_bar_chart` because you need to call `translate_options` as well. But I think you can still use `apply` to change the labels (e.g. `quest_frame[col] = quest_frame[col].apply(traslate_options)`) and then pass it in for the plotting.

Reply:
I translate the labels with `quest_frame[col] = quest_frame[col].apply(lambda x: traslate_options(x, opt_dict))` now